opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -400,7 +400,7 @@ def values(self) -> typing.ValuesView[str]:
 
 DEFAULT_TRACE_STATE = TraceState.get_default()
 _TRACE_ID_MAX_VALUE = 2**128 - 1
-_SPAN_ID_MAX_VALUE = 2**64 - 1
+_SPAN_ID_MAX_VALUE = 2**63 - 1
```

Comment:
Wouldn't this halve the value range of valid span IDs?
In the [proto](https://github.com/open-telemetry/opentelemetry-proto/blob/v1.0.0/opentelemetry/proto/trace/v1/trace.proto#L89-L95) defines it as 8 bytes, so it should be possible to fit the full (2^64)-1 range.
Additionally, `_SPAN_ID_MAX_VALUE` is only used by `SpanContext.validate` but the span IDs are still produced by the [IdGenerator](https://github.com/open-telemetry/opentelemetry-python/blob/v1.21.0/opentelemetry-sdk/src/opentelemetry/sdk/trace/id_generator.py#L49) with the full 64bit range. This would mean that potentially halve of the produced span IDs would be considered as invalid. 

Was the span ID per chance used as span attribute? In this case i think the exception would make sense since integer span attributes in the proto are defined as (signed) `int64` and in python the sign bit is stored separately from the value. 
The same would also apply for trace IDs, just that these are 128 bit wide.